File: pingdart-python/pingdart_sdk/services/whatsapp_service.py
class WhatsAppService:
    def __init__(self, http_client):
        self.http = http = http_client

    def initialize(self, client_id, phone_number):
        try:
            response = self.http.post("/whatsapp/initialize-whatsapp", json={
                "clientId": client_id,
                "phoneNumber": phone_number
            })
            response.raise_for_status()
            return response.json()
        except Exception as e:
            raise Exception(f"Failed to initialize WhatsApp: {str(e)}")

    def send_message(self, client_id, to, message=None, media_data=None, message_type='text', mimetype=None, filename=None):
        if not client_id or not to or (not message and not media_data):
            raise ValueError("client_id, to, and message/media_data are required")

        payload = {
            "clientId": client_id,
            "phoneNumber": to,
            "message": message,
            "type": message_type,
            "mediaData": media_data,
            "mimetype": mimetype,
            "filename": filename
        }
        
        try:
            response = self.http.post("/whatsapp/send-message", json=payload)
            response.raise_for_status()
            return response.json()
        except Exception as e:
            raise Exception(f"Failed to send WhatsApp message: {str(e)}")

    def check_status(self, client_id):
        try:
            response = self.http.post("/whatsapp/check-status", json={"clientId": client_id})
            response.raise_for_status()
            return response.json()
        except Exception as e:
            raise Exception(f"Failed to check WhatsApp status: {str(e)}")

File: pingdart-php/src/Services/WhatsAppService.php
<?php

namespace PingDart\SDK\Services;

use GuzzleHttp\Client;

class WhatsAppService {
    private $http;

    public function __construct(Client $http) {
        $this->http = $http;
    }

    public function initialize(string $clientId, string $phoneNumber) {
        try {
            $response = $this->http->post('whatsapp/initialize-whatsapp', [
                'json' => ['clientId' => $clientId, 'phoneNumber' => $phoneNumber]
            ]);
            return json_decode($response->getBody()->getContents(), true);
        } catch (\Exception $e) {
            throw new \Exception("Failed to initialize WhatsApp: " . $e->getMessage());
        }
    }

    public function sendMessage(array $params) {
        if (!isset($params['clientId'], $params['to']) || (!isset($params['message']) && !isset($params['mediaData']))) {
            throw new \InvalidArgumentException("clientId, to, and message/mediaData are required");
        }

        try {
            $response = $this->http->post('whatsapp/send-message', [
                'json' => [
                    'clientId' => $params['clientId'],
                    'phoneNumber' => $params['to'],
                    'message' => $params['message'] ?? null,
                    'type' => $params['type'] ?? 'text',
                    'mediaData' => $params['mediaData'] ?? null,
                    'mimetype' => $params['mimetype'] ?? null,
                    'filename' => $params['filename'] ?? null
                ]
            ]);
            return json_decode($response->getBody()->getContents(), true);
        } catch (\Exception $e) {
            throw new \Exception("Failed to send WhatsApp message: " . $e->getMessage());
        }
    }

    public function checkStatus(string $clientId) {
        try {
            $response = $this->http->post('whatsapp/check-status', [
                'json' => ['clientId' => $clientId]
            ]);
            return json_decode($response->getBody()->getContents(), true);
        } catch (\Exception $e) {
            throw new \Exception("Failed to check WhatsApp status: " . $e->getMessage());
        }
    }
}
